usaspending_api/common/recipient_lookups.py

```diff
@@ -0,0 +1,59 @@
+from usaspending_api.recipient.models import RecipientLookup
+
+
+def obtain_recipient_uri(recipient_name, recipient_unique_id, parent_recipient_unique_id=None):
+    if not recipient_unique_id:
+        return create_recipient_uri_without_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id)
+
+    return fetch_recipient_hash_using_name_and_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id)
+
+
+def create_recipient_uri_without_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id=None):
+    recipient_hash = generate_missing_recipient_hash(recipient_name, recipient_unique_id)
+    recipient_level = obtain_recipient_level({"duns": recipient_unique_id, "parent_duns": parent_recipient_unique_id})
+    return combine_recipient_hash_and_level(recipient_hash, recipient_level)
+
+
+def fetch_recipient_hash_using_name_and_duns(recipient_name, recipient_unique_id, parent_recipient_unique_id):
```

Comment:
`recipient_name` isn't used in this function which also makes the function name a bit misleading.

Reply:
Why `fetch_recipient_HASH_using_name_and_duns` and its sister function called `create_recipient_URI_without_duns`?  Should they both be URI?

Reply:
fixed